pkg/model/dynamic-plugins.go

```diff
@@ -203,6 +197,90 @@ func (p *DynamicPlugins) pluginsFromConfigMap() ([]DynaPlugin, error) {
 	return pluginsConfig.Plugins, nil
 }
 
+func (p *DynamicPlugins) mergeWith(specData string) (string, error) {
+
+	if p.ConfigMap == nil {
+		return "", fmt.Errorf("dynamic plugins ConfigMap is not set")
+	}
+	modelData := p.ConfigMap.Data[DynamicPluginsFile]
+	var modelPluginsConfig, specPluginsConfig DynaPluginsConfig
+	if modelData != "" {
+		if err := yaml.Unmarshal([]byte(modelData), &modelPluginsConfig); err != nil {
+			return "", fmt.Errorf("failed to unmarshal model ConfigMap data: %w", err)
```

Comment:
Should we make this error more readable e.g.:
```suggestion
			return "", fmt.Errorf("failed to unmarshal model ConfigMap data for key '%s': %w", 
    DynamicPluginsFile, err)
```

Reply:
No real needs, as only one known key expected (as documented).